packages/filebrowser/src/browser.ts

```diff
@@ -376,6 +477,18 @@ export class FileBrowser extends SidePanel {
     return this.listing.modelForClick(event);
   }
 
+  toggleFileFilter(): void {
+    const oldValue = this.showFileFilter;
+    this.showFileFilter = !oldValue;
+    // Toggle button state for the filter button
+    (this.toolbar.layout as PanelLayout).widgets.forEach(widget => {
+      const button = widget as ToolbarButton;
+      if (button.node.getAttribute('data-jp-item-name') === 'toggleFilter') {
+        button.pressed = !oldValue;
+      }
+    });
```

Comment:
Instead of directly modifying the DOM, can you use the command approach discussed earlier? I would suggest using `isToggled` with custom styling if needed. It adds `lm-mod-toggled` as implemented here:

https://github.com/jupyterlab/jupyterlab/blob/ae123cb82015540804964f76ba4e56f21799ea27/packages/ui-components/src/components/toolbar.tsx#L1076-L1081

Reply:
Maybe `isToggled` in context of a button should in fact cause `pressed` state to be changed? Let me know your thoughts, we can always modify `packages/ui-components/src/components/toolbar.tsx` if needed.